site/docs/master/faq.md

```diff
@@ -0,0 +1,219 @@
+If you decide to take this approach, you must ensure that you are reading the version of the test code that corresponds to your test image.
+You can verify which version of the test image was used by inspecting the plugin definition which is available in the results tarball in `plugins/e2e/definition.json` under the key `Definition.spec.image`.
+For example, if the test image was `gcr.io/google-containers/conformance:v1.15.3`, you should read the code at the corresponding [v1.15.3 tag in GitHub][kubernetes-1.15.3].
+All the tests can be found within the `test/e2e` directory in the Kubernetes repository.
+
+### How can I run the E2E tests with certain test framework options set? What are the available options?
+How you provide options to the E2E test framework and determining which options you can set depends on which version of Kubernetes you are testing.
+
+To view the available options that you can set when running the tests, you can run the test executable for the conformance image you will be using as follows:
+
+```
+KUBE_VERSION=<Kubernetes version you are using>
+docker run -it gcr.io/google-containers/conformance:$KUBE_VERSION ./e2e.test --help
+```
+
+If you are running Kubernetes v1.16.0 or greater, a new feature was included in this release which makes it easier to specify your own options.
+This new feature allows arbitrary options to be specified when the tests are invoked.
+To use this, you must ensure the environment variable `E2E_USE_GO_RUNNER=true` is set.
+This is the default behavior from Sonobuoy v0.16.1 in the CLI and only needs to be manually set if working with a Sonobuoy manifest generated by an earlier version.
+If this is enabled, then you can provide your options with the flag `--plugin-env=e2e.E2E_EXTRA_ARGS`.
+For example, the following allows you set provider specific flags for running on GCE:
+
+```
+sonobuoy run --plugin-env=e2e.E2E_USE_GO_RUNNER=true \
+  --plugin-env=e2e.E2E_PROVIDER=gce \
+  --plugin-env=e2e.E2E_EXTRA_ARGS="--gce-zone=foo --gce-region=bar"
+```
+
+Before this version, it was necessary to build your own custom image which could execute the tests with the desired options.
+
+For details on the two different approaches that you can take, please refer to [our blog post][custom-e2e-image] which describes in more detail how to use the new v1.16.0 Go test runner and how to build your own custom images.
+
+
+### Some of the registries required for the tests are blocked with my test infrastructure. Can I still run the tests?
+Yes! Sonobuoy can be configured to use custom registries so that you can run the tests in airgapped environments.
+
+For more information and details on how to configure your environment, please refer to [our documentation for custom registries and air-gapped environments][airgap].
+
+### We have some nodes with custom taints in our cluster and the tests won't start. How can I run the tests?
```

Comment:
I'm dont think the podspec customization is the answer to this question.

So there are two use cases:
 - User has some nodes tainted and keep seeing messages about how the e2e tests are waiting for all the nodes to be schedulable (which doesnt happen) and then it times out.
 - User has all the nodes tainted so the same issue as the above happens, but even if they solve that issue, then all the jobs that the e2e tests try and make will not be schedulable and so tests are still not runnable.

In the first case, the upstream PR https://github.com/kubernetes/kubernetes/pull/81043 added a flag to allow the e2e framework to ignore certain taints when checking 'are all nodes ready to start tests'

In the 2nd case, an issue is still open to address this but will require that the e2e framework create a mutating webhook so that all jobs will tolerate certain taints when launched.

I think the podspec answer you gave answers a separate set of issues:
 - I have a tainted environment and I can't get my plugin to launch on any nodes, can I still run it?

and obviously it also answers the questions:
 - I need certain podspec options (like hostPID) set on my plugin, is that possible?
 - I need to run my plugin only on certain nodes, is that possible?

Reply:
Yep, you're totally right. Sorry, I forgot about that PR and was only thinking about it launching the actual plugin pods, not all the other things that e2e framework launches itself.

Reply:
I've updated this answer but it's likely that I am missing some information/context that would be helpful to mention here. Let me know what you think.